src/qt/openuridialog.cpp

```diff
@@ -63,6 +64,30 @@ void OpenURIDialog::on_selectFileButton_clicked()
     QString filename = GUIUtil::getOpenFileName(this, tr("Select payment request file to open"), "", "", NULL);
     if (filename.isEmpty())
         return;
-    QUrl fileUri = QUrl::fromLocalFile(filename);
-    ui->uriEdit->setText("pivx:?r=" + QUrl::toPercentEncoding(fileUri.toString()));
+
+    QFile file(filename);
+    if(!file.exists()) {
+        inform(tr("File not found"));
+        return;
+    }
+
+    if (file.open(QIODevice::ReadOnly | QIODevice::Text)){
+        QByteArray r = file.readAll();
+        if (r.size() > 200) {
+            inform(tr("Parsed data too large"));
+            return;
+        }
+        QString str = QString::fromStdString(r.toStdString());
```

Comment:
This is non-op on Qt versions prior to 5.4: (`error: ‘class QByteArray’ has no member named ‘toStdString’`). We currently support down to Qt 5.0.

While I think we should certainly look at bumping our minimum supported Qt version, and have discussed doing just that, it is not going to happen for 4.0.0.

Reply:
Ok, doing the changes. What an old and useless QT versions that we are currently supporting.

Reply:
Done.